pkg/domain/infra/abi/images.go

```diff
@@ -54,13 +63,11 @@ func (ir *ImageEngine) Prune(ctx context.Context, opts entities.ImagePruneOption
 		WithSize:                true,
 	}
 
-	if !opts.All {
+	if !opts.All && !filterPrefix("dangling=", pruneOptions.Filters) {
 		pruneOptions.Filters = append(pruneOptions.Filters, "dangling=true")
 	}
-	if opts.External {
+	if opts.External && !filterPrefix("containers=", pruneOptions.Filters) {
 		pruneOptions.Filters = append(pruneOptions.Filters, "containers=external")
-	} else {
-		pruneOptions.Filters = append(pruneOptions.Filters, "containers=false")
```

Comment:
Please remove the change here and also remove the `filterPrefix` function above

Reply:
We are blocking double duplicate filters.  So if it has containers=external from some were else, then we don't want to add it again.

Reply:
@vrothberg fixed this in c/common see the comments above, you have to rebase this to get the latest common